cypress-tests/cypress/e2e/v2/searchEntities.cy.ts

```diff
@@ -0,0 +1,180 @@
+          namespace: user.username,
+          slug: proj,
+        }).then((response) => {
+          // eslint-disable-line max-nested-callbacks
+          if (response.status >= 400) {
+            throw new Error("Failed to create project");
+          }
+        });
+      }
+      for (const grp of [groups.first, groups.second]) {
+        createGroupIfMissingAPI({
+          name: grp,
+          slug: grp,
+        }).then((response) => {
+          // eslint-disable-line max-nested-callbacks
+          if (response.status >= 400) {
+            throw new Error("Failed to create group");
+          }
+        });
+      }
+    });
+  });
+
+  // Restore the session (login)
+  beforeEach(() => {
+    cy.session(
+      sessionId,
+      () => {
+        cy.robustLogin();
+      },
+      validateLoginV2,
+    );
+  });
+
+  // Cleanup the resources after the test
+  after(() => {
+    [projects.first, projects.second].forEach((proj) => {
+      getProjectByNamespaceAPIV2({
+        slug: proj,
+        namespace: userNamespace,
+      }).then((response) => {
+        if (response.status === 200) {
+          deleteProjectFromAPIV2({
+            id: response.body.id,
+            slug: proj,
+          });
+        }
+      });
+    });
+    [groups.first, groups.second].forEach((grp) => {
+      getGroupFromAPI(grp).then((response) => {
+        if (response.status === 200) {
+          deleteGroupFromAPI(grp);
+        }
+      });
+    });
+  });
+
+  it("Can search for entities and filter works", () => {
+    cy.visit("/v2");
+
+    // Search for string
+    cy.getDataCy("navbar-link-search").click();
+    cy.intercept(
+      new RegExp(
+        `(?:/ui-server)?/api/search/query\\?q=.*?${stringRandomOne}.*`,
+      ),
+    ).as("searchQueryOne");
+    cy.getDataCy("search-input").clear().type(stringRandomOne);
+    cy.getDataCy("search-button").click();
+    cy.wait("@searchQueryOne");
+    cy.getDataCy("search-card").should("have.length", 2);
+    cy.intercept(
+      new RegExp(
+        `(?:/ui-server)?/api/search/query\\?q=.*?${stringRandomTwo}.*`,
+      ),
+    ).as("searchQueryTwo");
+    cy.getDataCy("search-input").clear().type(stringRandomTwo);
+    cy.getDataCy("search-button").click();
+    cy.wait("@searchQueryTwo");
+    cy.getDataCy("search-card").should("have.length", 2);
+
+    // Filter for projects and groups
+    cy.getDataCy("search-filter-type-group").filter(":visible").click();
+    cy.wait("@searchQueryTwo");
+    cy.getDataCy("search-card").should("have.length", 1);
+
+    cy.getDataCy("search-input").clear().type(stringRandomOne);
+    cy.getDataCy("search-button").click();
+    cy.wait("@searchQueryOne");
+    cy.getDataCy("search-card").should("have.length", 1);
+
+    cy.getDataCy("search-filter-type-group").filter(":visible").click();
+    cy.getDataCy("search-filter-type-project").filter(":visible").click();
+    cy.wait("@searchQueryOne");
+    cy.getDataCy("search-card").should("have.length", 1);
```

Comment:
suggestion: Same as above - and in this case it is even easier because there is only one result. Lets assert that we got the right projects or groups back and not something else.

Reply:
I now more precisely check for group and project names wherever I check the length 893dff814f5af245d8ba87291c206f98f313f061